File: Prototype_node/routes/modules/MatchedTransaction.js
function MatchedTransaction(bid, ask, date, price) {
	this.bid = bid;
	this.ask = ask;
	this.date = date;
	this.price = price;
	this.stock = bid.getStock();
}

MatchedTransaction.prototype.getStock = function() {
	return this.stock;
}

MatchedTransaction.prototype.getPrice = function() {
	return this.price;
}

MatchedTransaction.prototype.getDate = function() {
	return this.date;
}

MatchedTransaction.prototype.getBuyerId = function() {
	return this.bid.getUserId();
}

MatchedTransaction.prototype.getSellerId = function() {
	return this.ask.getUserId();
}

MatchedTransaction.prototype.toString = function() {
	var dateString = [[AddZero(this.date.getDate()), AddZero(this.date.getMonth() + 1), this.date.getFullYear()].join("/"), [AddZero(this.date.getHours()), AddZero(this.date.getMinutes())].join(":"), this.date.getHours() >= 12 ? "PM" : "AM"].join(" ");
	return "stock: " + this.stock + ", amt: " + this.price + ", bidder userId: " + this.bid.getUserId() + ", seller userId: " + this.ask.getUserId() + ", date: " + dateString;
}

//Pad given value to the left with "0"
function AddZero(num) {
    return (num >= 0 && num < 10) ? "0" + num : num + "";
}

module.exports.MatchedTransaction = MatchedTransaction;

File: Prototype_Apache1/src/java/aa/MatchedTransaction.java
package aa;

import java.util.Date;

// represents a matched bid and ask
public class MatchedTransaction {

  private Bid bid;
  private Ask ask;
  private Date date;
  private int price;
  private String stock;

  // constructor
  public MatchedTransaction(Bid b, Ask a, Date d, int p) {
    this.bid = b;
    this.ask = a;
    this.date = d;
    this.price = p;
    this.stock = b.getStock(); // or a.getStock(). will be the same
  }

  // getter
  public String getStock() {
    return stock;
  }

  public int getPrice() {
    return price;
  }

  public Date getDate() {
    return date;
  }

  public String getBuyerId() {
    return bid.getUserId();
  }

  public String getSellerId() {
    return ask.getUserId();
  }

  // toString
  public String toString() {
    return "stock: " + stock + ", amt: " + price + ", bidder userId: " + bid.getUserId() + ", seller userId: " + ask.getUserId() + ", date: " + date;
  }
}
